index.js

```diff
@@ -109,12 +109,24 @@ module.exports = function(_version) {
                 // NOOP, since no special logic for that type
             }
 
+            // Decide way_name with special handling for name and ref
+            var wayName;
+            var name = (step.name || '').replace(' (' + step.ref + ')', ''); // Remove hack from Mapbox Directions mixing ref into name
```

Comment:
Use a regular expression to make sure we only replace this substring at the end of the name.